ingestion/src/metadata/ingestion/api/topology_runner.py

```diff
@@ -197,39 +198,19 @@ def yield_and_update_context(
         The default implementation is based on a get_by_name validation
         """
 
-        entity = None
-
-        entity_fqn = self.fqn_from_context(stage=stage, entity_request=right)
-
-        # we get entity from OM if we do not want to overwrite existing data in OM
-        if not stage.overwrite and not self._is_force_overwrite_enabled():
-            entity = self.metadata.get_by_name(
-                entity=stage.type_,
-                fqn=entity_fqn,
-                fields=["*"],  # Get all the available data from the Entity
-            )
-        # if entity does not exist in OM, or we want to overwrite, we will yield the entity_request
-        if entity is None:
-            tries = 3
-            while not entity and tries > 0:
-                yield entity_request
-                entity = self.metadata.get_by_name(
-                    entity=stage.type_,
-                    fqn=entity_fqn,
-                    fields=["*"],  # Get all the available data from the Entity
-                )
-                tries -= 1
+        entity_name = model_str(right.name)
+        yield entity_request
 
         # We have ack the sink waiting for a response, but got nothing back
-        if stage.must_return and entity is None:
+        if stage.must_return and entity_name is None:
```

Comment:
does it make sense to have the `must_return` anymore? We are not running any GET so we might as well skip this validation I guess.

Another approach which I might like a bit better is to keep a `must_return` only for important pieces, such as the Services. Those we want to be able to GET all the time otherwise there's no point continuing the ingestion

Reply:
Added a get_by_name call only if the `must_return` attribute is kept true